Topics/System_Design/Authentication_Service_Design.md

```diff
@@ -0,0 +1,85 @@
+# Introduction: Designing an Authentication Service
+
+**Authentication** is the first gateway users pass through when using any digital service to prove their identities. **Authorization** is how we ensure the *right people* are accessing the *right data*.
+A well-designed authentication and authorization service has to strike the correct balance of security and user convenience. This balance shifts depending on what service you're offering; an in-browser casual multiplayer game will have completely different needs for security and user convenience compared to governmental corporate software.
+
+## Table of Contents
+
+1. [Types of Authentication](#auth-types)
+	1. [Multi-Factor Authentication](#mfa)
+	2. [Standardized Protocols](#protocols)
+2. [Sessions and Tokens](#sessions-tokens)
+	1. [Designing Token-Based Authentication](#token-design)
+4. [Design Considerations](#auth-design)
+5. [Common Technologies](#auth-common-tech)
+6. [Further Reading](#further-reading)
+
+## Types of Authentication <a name="auth-types"></a>
+
+We all know about usernames and passwords, or **single-factor authentication**, which grants access to a system when the credentials are correct. However, in most modern systems, this is no longer the case for several reasons:
+1. Users don't keep their passwords secure
+2. Users are vulnerable to phishing schemes
+3. Users easily forget their own credentials
+4. Credentials can be easily cracked
+
+A common theme we'll see is that authentication schemes will try to work around the weakest security link in the system, which are the users themselves. While small improvements can help, like security questions or increasing requirements for password hygiene (minimum N characters, with at least M special characters), modern approaches have better ways of working around this limitation. 
```

Comment:
I am a little confused as to what "this limitation" is referring to. Is it referring to single-factor authentication limitations? Or perhaps the limitations of making the small improvements?

Reply:
Good catch, I'll make it clear! This is about users being the weakest link.